From c55f8011cddb697f0d0310b4c8102850a8749887 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:08:47 -0400
Subject: [usb] usb_wwan: kill interrupt urb explicitly at suspend

Message-id: <1410887463-20674-456-git-send-email-dzickus@redhat.com>
Patchwork-id: 95461
O-Subject: [RHEL7 PATCH 455/591] USB: usb_wwan: kill interrupt urb explicitly at suspend
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit ae75c940181c3d4044f7b1adf07ada877ad9cb83
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:23:24 2014 +0200

    USB: usb_wwan: kill interrupt urb explicitly at suspend

    As the port interrupt URB is submitted by the subdriver at open, we
    should also kill it explicitly at suspend (even though this will be
    taken care of by USB serial core otherwise).

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/usb_wwan.c b/drivers/usb/serial/usb_wwan.c
index c4a815c..b671d59 100644
--- a/drivers/usb/serial/usb_wwan.c
+++ b/drivers/usb/serial/usb_wwan.c
@@ -562,7 +562,7 @@ int usb_wwan_port_remove(struct usb_serial_port *port)
 EXPORT_SYMBOL(usb_wwan_port_remove);
 
 #ifdef CONFIG_PM
-static void stop_read_write_urbs(struct usb_serial *serial)
+static void stop_urbs(struct usb_serial *serial)
 {
  int i, j;
  struct usb_serial_port *port;
@@ -578,6 +578,7 @@ static void stop_read_write_urbs(struct usb_serial *serial)
    usb_kill_urb(portdata->in_urbs[j]);
   for (j = 0; j < N_OUT_URB; j++)
    usb_kill_urb(portdata->out_urbs[j]);
+  usb_kill_urb(port->interrupt_in_urb);
  }
 }
 
@@ -595,7 +596,7 @@ int usb_wwan_suspend(struct usb_serial *serial, pm_message_t message)
  intfdata->suspended = 1;
  spin_unlock_irq(&intfdata->susp_lock);
 
- stop_read_write_urbs(serial);
+ stop_urbs(serial);
 
  return 0;
 }
-- 
1.7.1